craft_store/auth.py

```diff
@@ -0,0 +1,87 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Craft Store Authentication Store."""
+
+import base64
+import logging
+
+import keyring
+import keyring.errors
+
+from . import errors
+
+logger = logging.getLogger(__name__)
+
+
+class Auth:
+    """Auth wraps around the keyring to store credentials.
+
+    The application_name and host are used as key/values in the keyring to set,
+    get and delete credentials.
+
+    Credentials are base64 encoded into the keyring and decoded on retrieval.
+
+    :ivar application_name: name of the application using this library.
+    :ivar host: specific host for the store used.
+    """
+
+    def __init__(self, application_name: str, host: str) -> None:
+        """Initialize Auth.
```

Comment:
`__init__` is always about initializing the class, and the parameters are documented in the class docstring, so why this duplication?

Reply:
I would think that this is not always 1:1. One is instance vars and the others are initialization params.

Reply:
By the way, Sphinx used to do the wrong thing when class init parameters are documented in the `__init__()` method instead of directly in the class docstring, I suggest testing this to see if it's rendered correctly.

Reply:
changed to credentials in https://github.com/canonical/craft-store/pull/5/commits/337c2f2faeba6c259691d460d3c4efff0c0a4d7b